R/estimateAlpha.R

```diff
@@ -0,0 +1,248 @@
+                                           verbose=FALSE),
+                           FUN=FUN,
+                           args.fun=list(index=index, assay.type="subsampled"),
+                           ...,
+                           name=name)
+    } else {
+        suppressWarnings(do.call(FUN, args = c(list(x, assay.type=assay.type,
+                                                  assay_name=assay_name,
+                                                  index=index, name=name),
+                                                  list(...))))
+    }
+
+}
+
+## Helper functions
+
+.get_indices <- function(measure) {
+    switch(measure,
+           "diversity" = c("coverage_diversity", "coverage",
+                           "faith_diversity", "faith",
+                           "fisher_diversity", "fisher",
+                           "gini_simpson_diversity", "gini_simpson",
+                           "inverse_simpson_diversity", "inverse_simpson",
+                           "log_modulo_skewness_diversity", "log_modulo_skewness",
+                           "shannon_diversity", "shannon"),
+           "dominance" = c("absolute_dominance", "absolute",
+                           "dbp_dominance", "dbp",
+                           "core_abundance_dominance", "core_abundance",
+                           "gini_dominance", "gini", 
+                           "dmn_dominance", "dmn",
+                           "relative_dominance", "relative",
+                           "simpson_lambda_dominance", "simpson_lambda"),
+           "evenness" = c("camargo_evenness", "camargo",
+                          "pielou_evenness", "pielou",
+                          "simpson_evenness",
+                          "evar_evenness", "evar",
+                          "bulla_evenness", "bulla"),
+           "richness" = c("ace_richness", "ace",
+                          "chao1_richness", "chao1",
+                          "hill_richness", "hill",
+                          "observed_richness", "observed"))
+}
+
+.alpha_rarefaction <- function(x,
+                               nrounds=1L,
+                               seed=123,
+                               args.sub=list(assay.type="counts",
+                                             min_size=min(colSums(assay(x, "counts")),
+                                                          na.rm = TRUE),
+                                             verbose=FALSE),
+                               FUN=estimateDiversity,
+                               args.fun=c(index="shannon",
+                                             assay.type="subsampled"),
+                               ...,
+                               name = args.fun$index) {
+    set.seed(seed)
+    colData(x)[, name] <- lapply(seq(nrounds), function(i){
+        x_sub <- do.call(subsampleCounts, args = c(list(x), args.sub))
+        suppressWarnings(x_sub <- do.call(FUN, args = c(list(x_sub),
```

Comment:
Why warnings are suppressed? Add comment there. These should be avoided

Reply:
Due to the deprication message added to the alpha functions, it needed to be supressed when called from estimateAlpha to pass the test.